### PR TITLE
Update caddy import to use the one from coredns

### PR DESCRIPTION
**Pull request by cybertron**
When caddy moved to v2, coredns chose to stay with v1 by moving it
in-tree. Plugins need to update their imports to reflect this.

Note that this will fail the build-coredns ci job until the
openshift/coredns repo has been updated to the new version. I have
tested it locally with a newer version and it works.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,54 @@
...
+	go.uber.org/multierr v1.6.0 // indirect
+	go.uber.org/zap v1.16.0 // indirect
+	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83 // indirect
+	golang.org/x/net v0.0.0-20210222171744-9060382bd457
+	golang.org/x/oauth2 v0.0.0-20210220000619-9bb904979d93 // indirect
+	golang.org/x/term v0.0.0-20210220032956-6a3ed077a48d // indirect
+	golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	k8s.io/api v0.20.4 // indirect
+	k8s.io/client-go v11.0.0+incompatible // indirect
```
Comment (sgreene570) [inline go.mod:48]: is this version tag for client-go right?

Comment (cybertron) [reply go.mod]: It looks like I passed an errant -u to go get when I updated coredns. I think this should be fixed now.

**Comment by cybertron** [comment]
For reference, this is in relation to https://github.com/coredns/coredns/pull/4018

**Pull request merged**